crates/services/gas_price_service/src/v1/da_source_adapter/block_committer_costs.rs

```diff
@@ -6,57 +9,334 @@ use crate::v1::da_source_adapter::{
+            if let Some(value) = &mut value {
+                value.sequence_number = seq_no;
+                value.blocks_range =
+                    value.blocks_range.end * seq_no..value.blocks_range.end * seq_no + 10;
+                value.da_block_height = value.da_block_height + (seq_no + 1).into();
+                value.total_cost += 1;
+                value.total_size_bytes += 1;
+            }
+            Ok(value)
+        }
+        async fn get_bundles_by_range(
+            &self,
+            _: core::ops::Range<u64>,
+        ) -> DaBlockCostsResult<Vec<Option<RawDaBlockCosts>>> {
+            Ok(vec![self.value.clone()])
+        }
+    }
+
+    fn test_da_block_costs() -> RawDaBlockCosts {
+        RawDaBlockCosts {
+            sequence_number: 1,
+            blocks_range: 0..10,
+            da_block_height: 1u64.into(),
+            total_cost: 1,
+            total_size_bytes: 1,
+        }
+    }
+
+    #[tokio::test]
+    async fn request_da_block_cost__when_last_value_is_none__then_get_latest_costs_is_called(
+    ) {
+        // given
+        let da_block_costs = test_da_block_costs();
+        let mock_block_committer =
+            MockBlockCommitterApi::new(Some(da_block_costs.clone()));
+        let mut block_committer =
+            BlockCommitterDaBlockCosts::new(mock_block_committer, None);
+
+        // when
+        let block_committer_da_block_costs =
+            block_committer.request_da_block_cost().await.unwrap();
+
+        // then
+        assert_eq!(
+            block_committer_da_block_costs,
+            DaBlockCosts {
+                l2_block_range: da_block_costs.blocks_range.clone(),
+                blob_size_bytes: da_block_costs.total_size_bytes,
+                blob_cost_wei: da_block_costs.total_cost,
+            }
+        );
+        assert!(block_committer.last_value.is_some());
+    }
+
+    #[tokio::test]
+    async fn request_da_block_cost__when_last_value_is_some__then_get_costs_by_seqno_is_called(
+    ) {
+        // given
+        let mut da_block_costs = test_da_block_costs();
+        let mock_block_committer =
+            MockBlockCommitterApi::new(Some(da_block_costs.clone()));
+        let mut block_committer = BlockCommitterDaBlockCosts::new(
+            mock_block_committer,
+            Some(da_block_costs.clone()),
+        );
+
+        // when
+        let initial = block_committer.request_da_block_cost().await.unwrap();
+        let updated = block_committer.request_da_block_cost().await.unwrap();
+
+        // then
+        // assert_eq!(initial.blob_size_bytes, updated.blob_size_bytes);
+        // assert_eq!(initial.blob_cost_wei, updated.blob_cost_wei);
+        assert_ne!(initial.l2_block_range, updated.l2_block_range);
+    }
+
+    #[tokio::test]
+    async fn request_da_block_cost__when_response_is_none__then_error() {
+        // given
+        let mock_block_committer = MockBlockCommitterApi::new(None);
```

Comment:
These names are confusing. Maybe `mock_api` to differentiate it from the `block_committer`.

This applies to all the tests.